src/client/tokens.js

```diff
@@ -0,0 +1,315 @@
+const msal = require('@azure/msal-node')
+const fs = require('fs')
+const path = require('path')
+const XboxLiveAuth = require('@xboxreplay/xboxlive-auth')
+const fetch = require('node-fetch')
+
+function debug(...message) {
+  console.debug(message[0])
+}
+
+// Manages Microsoft account tokens
+class MsaTokenManager {
+  constructor({ msalConfig, scopes, cacheLocation }) {
+    this.msaClientId = msalConfig.auth.clientId
+    this.scopes = scopes
+
+    if (!cacheLocation) cacheLocation = path.join(__dirname, './msa-cache.json')
```

Comment:
Maybe assign cacheLocation to each of its respective classes like so? I fear we might bleed into other classes.

`this.cacheLocation = cacheLocation ? cacheLocation : path.join(`${__dirname}/msa_cache.json`)`